internal/framework/provider/functions/manifest_decode_multi_test.go

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package functions_test
+
+import (
+	"fmt"
+	"math/big"
+	"os"
+	"path"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/knownvalue"
+	"github.com/hashicorp/terraform-plugin-testing/statecheck"
+)
+
+func TestManifestDecodeMulti(t *testing.T) {
+	t.Parallel()
+
+	outputName := "test"
+
+	resource.UnitTest(t, resource.TestCase{
+		ProtoV5ProviderFactories: testAccProtoV5ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testManifestDecodeMultiConfig("testdata/decode_single.yaml"),
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownOutputValue(outputName, knownvalue.ListExact([]knownvalue.Check{
+						knownvalue.ObjectExact(map[string]knownvalue.Check{
+							"apiVersion": knownvalue.StringExact("v1"),
+							"data": knownvalue.ObjectExact(map[string]knownvalue.Check{
+								"configfile": knownvalue.StringExact("---\ntest: document\n"),
+							}),
+							"kind": knownvalue.StringExact("ConfigMap"),
+							"metadata": knownvalue.ObjectExact(map[string]knownvalue.Check{
+								"labels": knownvalue.ObjectExact(map[string]knownvalue.Check{
+									"test": knownvalue.StringExact("test---label"),
+								}),
+								"name": knownvalue.StringExact("test-configmap"),
+							}),
+						}),
+					})),
+				},
+			},
+			{
+				Config: testManifestDecodeMultiConfig("testdata/decode_multi.yaml"),
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownOutputValue(outputName, knownvalue.ListExact([]knownvalue.Check{
```

Comment:
Curious, did you run into this panic bug? (since this is returning a tuple): https://github.com/hashicorp/terraform-plugin-testing/issues/310

I'm wondering if your tuple elements have equivalent types and that's why you didn't run into a panic here 🤔 

Either way, just an FYI that this bug does exist and is fixed, planned for release in `terraform-plugin-testing v1.8.0`

Reply:
I actually didn't run into this panic.. and my test case tuple has a bunch of objects of different types in it 🤔

Reply:
Oh! Coming back to this, I think the testing panic bug was really only for tuples with "primitives" as the elements. Looking back it's completely fine with tuples of objects. Apologies for the confusion!